### DISCUSSION TITLE
TSP Toolkit VSCode Extension: Multiple 'Undefined global' Warnings

### DISCUSSION THREAD

**Discussion by ktyubeshi** [tektronix/tsp-toolkit#70] [Q&A] [answered]
After installing the TSP Toolkit in VSCode and opening a sample TSP script (BatteryDischarge_20Dec2017.tsp), I'm getting numerous warning messages like "Undefined global 'smu'" as shown in this screenshot.

Is there any additional configuration needed?

![image](https://github.com/user-attachments/assets/260857db-dda6-4792-a87f-f0d97d5efb7f)

**Comment by jharajeev55** [comment, marked as answer]
@ktyubeshi please follow steps mentioned in readme(https://github.com/tektronix/tsp-toolkit?tab=readme-ov-file#configure-your-project)  to configure your SMU model,

**Reply by ktyubeshi** [reply]
@jharajeev55

Thank you. I've resolved the issue!
I understand that when I configure the instrument model, the TSP Toolkit automatically updates the `Lua.workspace.library` settings in `.vscode/settings.json`, which enables IntelliSense to work properly.
